mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulPlayScreenTest.kt

```diff
@@ -0,0 +1,123 @@
+import ch.epfl.sdp.mobile.state.StatefulPlayScreen
+import ch.epfl.sdp.mobile.test.infrastructure.persistence.auth.emptyAuth
+import ch.epfl.sdp.mobile.test.infrastructure.persistence.store.buildStore
+import ch.epfl.sdp.mobile.test.infrastructure.persistence.store.document
+import ch.epfl.sdp.mobile.test.infrastructure.persistence.store.emptyStore
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.filterIsInstance
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.test.runTest
+import org.junit.Rule
+import org.junit.Test
+
+class StatefulPlayScreenTest {
+  @get:Rule val rule = createComposeRule()
+
+  @Test
+  fun given_PlayScreen_When_ClickingNewGame_Then_LocalAndOnlinePlayAreDisplayed() = runTest {
+    val auth = emptyAuth()
+    val store = emptyStore()
+    val facade = AuthenticationFacade(auth, store)
+    val social = SocialFacade(auth, store)
+    val chess = ChessFacade(auth, store)
+
+    facade.signUpWithEmail("user1@email", "user1", "password")
+    val currentUser = facade.currentUser.filterIsInstance<AuthenticatedUser>().first()
+
+    val strings =
+        rule.setContentWithLocalizedStrings {
+          ProvideFacades(facade, social, chess) {
+            StatefulPlayScreen(
+                user = currentUser,
+                navigateToPrepareGame = {},
+                navigateToLocalGame = {},
+            )
+          }
+        }
+
+    rule.onNodeWithText(strings.newGame).performClick()
+    rule.onNodeWithText(strings.prepareGamePlayLocal).assertExists()
+    rule.onNodeWithText(strings.prepareGamePlayOnline).assertExists()
+  }
+
+  @Test
+  fun given_PlayScreen_When_ClickingNewGameAndLocalPlay_Then_LocalGameCallbackIsCalled() = runTest {
+    val auth = emptyAuth()
+    val store = buildStore {
+      collection("users") { document("userId2", ProfileDocument(name = "user2")) }
+    }
+    val facade = AuthenticationFacade(auth, store)
+    val social = SocialFacade(auth, store)
+    val chess = ChessFacade(auth, store)
+
+    facade.signUpWithEmail("user1@email", "user1", "password")
+    val currentUser = facade.currentUser.filterIsInstance<AuthenticatedUser>().first()
+
+    val channel = Channel<Unit>(capacity = 1)
+    val strings =
+        rule.setContentWithLocalizedStrings {
+          ProvideFacades(facade, social, chess) {
+            StatefulPlayScreen(
+                user = currentUser,
+                navigateToPrepareGame = {},
+                navigateToLocalGame = {
+                  channel.trySend(Unit)
+                  channel.close()
+                },
+            )
+          }
+        }
+
+    rule.onNodeWithText(strings.newGame).performClick()
+    rule.onNodeWithText(strings.prepareGamePlayLocal).performClick()
+
+    assertThat(channel.tryReceive().getOrNull()).isEqualTo(Unit)
+  }
+
+  @Test
+  fun given_PlayScreen_When_ClickingNewGameAndOnlinePlay_Then_OnlineGameCallbackIsCalled() =
```

Comment:
Same.

Reply:
Fixed in [2a29349](https://github.com/epfl-SDP/android/pull/226/commits/2a293496a9dde6403f42d2a252ef838212eafbb3)